reinforcement_learning/agents/common/lazy_tabular_action_value.py

```diff
@@ -19,62 +18,43 @@ class LazyTabularActionValue(BaseActionValue):
     MAX_PEN_WIDTH = 4
 
     def __init__(self):
-        self.action_value_dict = dict()
+        self.action_value_dict = defaultdict(lambda: defaultdict(lambda: self._default_cell_value))
 
     def __getitem__(self, key: tuple):
         assert len(key) == 2, f"Invalid key: {key}, should be tuple(BaseState, BaseAction)..."
 
         state, action = key
-
-        # Lazy initialization
-        if not self.action_value_dict.get(state):
-            self.action_value_dict[state] = {action: self._initial_cell_value}
-        elif not self.action_value_dict[state].get(action):
-            self.action_value_dict[state][action] = self._initial_cell_value
-
         return float(self.action_value_dict[state][action])
 
     def __setitem__(self, key, value):
         assert len(key) == 2, f"Invalid key: {key}, should be tuple(BaseState, BaseAction)..."
 
         state, action = key
+        self.action_value_dict[state][action] = float(value)
 
-        if not self.action_value_dict.get(state):
-            self.action_value_dict[state] = dict()
+    def max(self, state):
+        expected_returns = self.action_value_dict[state].values()
+        return max(expected_returns) if expected_returns else self._default_cell_value
 
-        self.action_value_dict[state][action] = float(value)
+    def argmax(self, state):
+        actions = self.action_value_dict[state]
+        return {key for (key, value) in actions.items() if value == max(actions.values())}
 
-    def max_over_actions(self, state):
-        expected_returns = self.action_value_dict.get(state, dict()).values()
-        return max(expected_returns) if expected_returns else self._initial_cell_value
-
-    def argmax_over_actions(self, state):
-        max_value = float('-inf')
-        max_value_actions = set()
-        for action, value in self.action_value_dict.get(state, {}).items():
-            if value > max_value:
-                max_value = value
-                max_value_actions = {action}
-            elif value == max_value:
-                max_value_actions.add(action)
-        return max_value_actions
+    def action_returns(self, state):
+        return self.action_value_dict[state]
 
     @property
-    def _initial_cell_value(self):
+    def _default_cell_value(self):
         """
-        This function is used to initialize action value table,
-        despite the fact that it's lazy initialization.
+        This function is used specify the default action value
 
         Returns
         -------
         Float
-            Arbitrarily chosen, initial value of the action-value table cell.
+            Initial value of the action-value table cell.
```

Comment:
Why have you changed initial to default in the method's name, but you've lived "initial" in the docstring?

Reply:
My bad